agent/websockets/shim.go

```diff
@@ -132,26 +140,30 @@ const (
         if (typeof msg == 'string') {
           return false;
         }
-        if (Array.isArray(msg)) {
-          if (msg.length == 1) {
-            if (typeof msg[0] == 'string') {
-              return false;
-            }
-          }
-        }
-        return true;
+		if (Array.isArray(msg)) {
+		  if (msg.length == 1) {
+	  		if (typeof msg[0] == 'string') {
+  			  return false;
+			}
+		  }
+		}
+		return true;
       }
       self.convertMessagesAndPush = function(msgs) {
         for (var i = 0; i < msgs.length; i++) {
           if (self.needsConversion(msgs[i].msg)) {
             var blob = new Blob([msgs[i].msg]);
             var reader = new FileReader();
             reader.addEventListener("loadend", function() {
-              msgs[i].msg = [reader.result];
-              self.convertMessagesAndPush(msgs);
+            if (self.protocolVersion != 0 ) {
+                msgs[i].msg = [btoa(reader.result)];
+            } else {
+                msgs[i].msg = [reader.result];
+            }
+            self.convertMessagesAndPush(msgs);
             });
-            reader.readAsText(blob);
-            return;
+			reader.readAsText(blob);
+			return;
```

Comment:
Similarly, here the whitespace got mangled.

Reply:
Fixed